dev/src/index.ts

```diff
@@ -1007,6 +1025,21 @@ export class Firestore {
       });
   }
 
+  /**
+   * Terminates the Firestore client and closes all open streams.
+   * @return A Promise that resolves when the client is terminated.
+   */
+  async terminate(): Promise<void> {
+    await this._clientPool.terminate();
+    this._terminated = true;
+  }
+
+  private verifyNotTerminated(): void {
```

Comment:
I don't think this method is needed - the logic you added to ClientPool does this just as well, and it is used in all code paths (such as WriteBatch.commit()). I think you can remove this method and the _terminated state from this file.